app/src/org/commcare/activities/QueryRequestUiController.kt

```diff
@@ -407,6 +411,32 @@ class QueryRequestUiController(
         return Localizer.processArguments(displayData.name, arrayOf("")).trim { it <= ' ' }
     }
 
+    private fun setHintPlaceholder(promptView: View, queryPrompt: QueryPrompt) {
+        val hintText = getHintText(queryPrompt)
+        if (hintText != "") {
+            val promptHelpButton = promptView.findViewById<View>(R.id.prompt_hint_button)
+            promptHelpButton.visibility = View.VISIBLE
+
+            promptHelpButton.setOnClickListener {
+                queryRequestActivity.showAlertDialog(StandardAlertDialog.getBasicAlertDialog(
+                        queryRequestActivity,
+                        "Hint",
+                        hintText,
+                        DialogInterface.OnClickListener { dialog, id ->
+                            dialog.dismiss()
+                        }
+                ))
+            }
+        }
+    }
+
+    private fun getHintText(queryPrompt: QueryPrompt): String {
+        val displayData = queryPrompt.display.evaluate()
+        val hintText = displayData.hintText ?: return "";
+
+        return Localizer.processArguments(hintText, arrayOf("")).trim { it <= ' ' }
```

Comment:
I don't think we need to call `Localizer. processArguments` here as calling `queryPrompt.display.evaluate()` should evaluate any localised strings itself.